notional/debt-settlement.js

```diff
@@ -0,0 +1,183 @@
+   */
+  async onSubscribeForm(args) {
+    return [
+      {
+        type: 'input-number',
+        id: 'time-buffer',
+        label: 'Notification Time',
+        default: 48,
+        description: 'Notify me X hours before debt maturity date.',
+      },
+    ];
+  }
+
+  /**
+   * Get debt positions for given account.
+   * @param {string} address
+   * @param {number} maxMaturity
+   * @param {number|undefined} blockNumber
+   * @returns {DebtPosition[]}
+   */
+  async getAccountSettlableDebtFromSubgraph(
+    address,
+    maxMaturity,
+    blockNumber = undefined
+  ) {
+    const blockNumberStatement = blockNumber
+      ? `, block: {number: ${blockNumber}}`
+      : '';
+
+    const query = `
+      {
+          account(id: "${address.toLowerCase()}"${blockNumberStatement}) {
+              portfolio(where: {notional_lt: 0, assetType: fCash, maturity_lt: ${maxMaturity}}) {
+                  id
+                  currency {
+                      underlyingSymbol
+                  }
+                  notional
+                  maturity
+              }
+          }
+      }`;
+    const jsonEncodedResponse = await fetch(SUBGRAPH_URL, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ query }),
+    });
+    const response = await jsonEncodedResponse.json();
+
+    if (!!response.data?.account?.portfolio) {
+      /** @type {Account} */
+      const account = response.data.account;
+
+      return account.portfolio;
+    }
+    return [];
+  }
+
+  /**
+   * Builds a notification message letting the user know that their account has debt that will mature soon.
+   *
+   * @param {Asset[]} debtPositions
+   * @param {number} timeBufferHours
+   * @returns {}
+   */
+  buildNotification(debtPositions, timeBufferHours) {
+    if (debtPositions.length === 0) {
+      return [];
+    }
+    const notificationLines = [
+      `Your account has debt that will mature in ${timeBufferHours} hours. Settle the positions to avoid penalty.`,
+      'Positions at risk:',
+    ];
+    const idParts = [];
+    for (const debtPosition of debtPositions) {
+      const {
+        id,
+        currency: { underlyingSymbol },
+      } = debtPosition;
+      idParts.push(id);
+
+      const balance = new BN(debtPosition.notional).div(NOTIONAL_BALANCE_PRECISION);
+      const formattedBalance = amountFormatter.format(balance.toString());
+
+      const maturity = new Date(Number(debtPosition.maturity) * 1000);
+      const formattedMaturityMonth = maturity.toLocaleString('en-us', {month: 'short'});
+      const formattedMaturity = `${formattedMaturityMonth} ${maturity.getDate()} ${maturity.getFullYear()}`;
+
+      notificationLines.push(`${formattedBalance} ${underlyingSymbol} (${formattedMaturity})`
+      );
+    }
+    return {
+      uniqueId: idParts.join(''),
+      notification: notificationLines.join('\n'),
```

Comment:
Because notifications appear as also push notifications, "\n" wont show and instead they will be appended without any spacing.
Can you implement a shorter notification text without the line-breaks?

Reply:
Updated the notification to fit on one line. It will look like this:

```
Your account has debt that will mature in 48 hours. Close these positions to avoid penalty: -2.6M USDC (Mar 28 2022) | -30 WBTC (Mar 28 2022)
```